builtin/mainmenu/common.lua

```diff
@@ -93,30 +72,28 @@ function render_serverlist_row(spec, is_favorite)
 		end
 	end
 
-	local grey_out = not is_server_protocol_compat(spec.proto_min, spec.proto_max)
+	local grey_out = not spec.is_compatible
 
-	local details
-	if is_favorite then
-		details = "1,"
-	else
-		details = "0,"
-	end
+	local details = {}
 
-	if spec.ping then
-		local ping = spec.ping * 1000
-		if ping <= 50 then
-			details = details .. "2,"
-		elseif ping <= 100 then
-			details = details .. "3,"
-		elseif ping <= 250 then
-			details = details .. "4,"
+	if spec.lag or spec.ping then
+		local lag = (spec.lag or 0) * 1000 + (spec.ping or 0) * 250
```

Comment:
this definitely need some kind of revision, too
I can't find a single server on the list that is below 4 bars which makes the entire indicator useless